controllers/disruption_controller.go

```diff
@@ -153,8 +155,14 @@ func (r *DisruptionReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
-	if err := r.manageInstanceSelectorCache(instance); err != nil {
-		r.log.Errorw("error managing selector cache", "error", err)
+	if err := r.DisruptionsWatchersManager.RemoveAllOrphanWatchers(); err != nil {
+		r.log.Errorw("error during the deletion of orphan watchers", "error", err)
+	}
+
+	r.DisruptionsWatchersManager.RemoveAllExpiredWatchers()
```

Comment:
Matthieu mentioned this in the last review, but even though it doesn't return an error anymore, I think we can move this work out of the reconcile loop and into async code. The check for expired and orphaned watchers aren't specific to a given instance, and I don't think we need to run them everytime we reconcile. wdyt?